mqtt/mqtt-bridge/src/client.rs

```diff
@@ -184,6 +186,36 @@ impl BridgeIoSource {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct InFlightPublishHandle {
+    publish_handle: PublishHandle,
+    permits: Arc<Semaphore>,
+}
+
+impl InFlightPublishHandle {
+    fn new(publish_handle: PublishHandle, permits: Arc<Semaphore>) -> Self {
+        Self {
+            publish_handle,
+            permits,
+        }
+    }
+
+    pub async fn publish(&self, publication: Publication) {
+        let mut publish_handle = self.publish_handle.clone();
+        let permits = self.permits.clone();
```

Comment:
We can't pass once acquired because it contains a reference to the original semaphore with a lifetime 'a.